base/utils/R/read.output.R

```diff
@@ -16,160 +16,241 @@
+      "in directory:", outdir, "including files", basename(ncfiles)
+    )
   }
-  
+
   result <- list()
 
-  if (!nofiles) {
+  if (nofiles) {
+    PEcAn.logger::logger.info("No files found. Returning all NA.")
+    result <- lapply(variables, function(x) NA)
+  } else {
     for (ncfile in ncfiles) {
+      if (verbose) PEcAn.logger::logger.debug("Processing file: ", ncfile)
       nc <- ncdf4::nc_open(ncfile)
-      if(dataframe==TRUE){
-        seconds <- udunits2::ud.convert(nc$dim$time$vals, nc$dim$time$units, paste("seconds since", run_origin))
+      if (is.null(variables)) {
+        variables <- names(nc[["var"]])
+        PEcAn.logger::logger.info(
+          "Variables not specified. Reading output for all variables, which are as follows: ",
+          paste(variables, collapse = ", ")
+        )
+      }
+
+      if (dataframe) {
+        seconds <- udunits2::ud.convert(
+          nc$dim$time$vals,
+          nc$dim$time$units,
+          paste("seconds since", run_origin)
+        )
         result[["posix"]] <- abind::abind(result[["posix"]], seconds)
       }
       for (v in variables) {
-        if (v %in% c(names(nc$var), names(nc$dim))) {
-          newresult <- ncdf4::ncvar_get(nc, v)
-          # begin per-pft read
-          # check if the variable has 'pft' as a dimension
-          if("pft" %in% sapply(nc$var[[v]]$dim, `[[`, "name")){
-            # means there are PFT specific outputs we want
-            # the variable *PFT* in standard netcdfs has *pft* dimension, 
-            # numbers as values, and full pft names as an attribute
-            # parse pft names and match the requested
-            pft.string <- ncdf4::ncatt_get(nc, "PFT")
-            pft.ind <- strsplit(pft.string$long_name, ",")[[1]] == pft.name
-            # dimensions can differ from model to model or run to run
-            # there might be other cases that are not covered here
-            dim.check <- length(dim(newresult))
-            if(any(pft.ind)){ # means pft.name passed, we want to read pft-specific outputs
-              if(dim.check == 1){
-                newresult <- newresult[pft.ind] 
-              }else{
-                newresult <- newresult[,pft.ind] 
-              }
-            }else{ 
-              # means this variable is available as per-pft, so written as such to standard ncdf files
-              # but we still want to read as total
-              if(dim.check == 1){
-                newresult <- sum(newresult)
-              }else{
-                newresult <- apply(newresult,1,sum)
-              }
-            }
-          } # end of per-pft read
-          
-          # Dropping attempt to provide more sensible units because of graph unit errors,
-          # issue #792 if(v %in% c(cflux, wflux)){ newresult <- udunits2::ud.convert(newresult, 'kg
-          # m-2 s-1', 'kg ha-1 yr-1') }
-          result[[v]] <- abind::abind(result[[v]], newresult)
-        } else if (!(v %in% names(nc$var))) {
```

Comment:
This code is basically the same -- I just switched the `if-else` cases so the shorter one came first. In general, I think this:

```r
if (something) {
  # Quick evaluation
  ...
} else {
  # Longer, more involved evaluation
  ...
  ...
  ...
  ...
  ...
}
```

...is more readable than the reverse:

```r
if (!something) {
  # Longer, more involved case
  ...
  ...
  ...
  ...
} else {
  # Short case
  ...
}
```

In this specific case, I also eliminate an `else` block by using an `if (...) next`, which skips the rest of the body and jumps to the next iteration.